src/test/projects/include-exclude/pom.xml

```diff
@@ -0,0 +1,51 @@
+<!--
+
+    Copyright 2021 the original author or authors.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.apache.maven.caching.test.include-exclude</groupId>
+    <artifactId>include-exclude</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+    <packaging>jar</packaging>
+
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <!-- Testing the project properties to include / exclude -->
+
+        <!-- one more file and one more folder to scan -->
+        <maven.build.cache.input.extraFile>extraFile.txt</maven.build.cache.input.extraFile>
+        <maven.build.cache.input.extraFolder>second_folder_outside_src</maven.build.cache.input.extraFolder>
+        <!-- Exclude directory in an extra included folder  -->
+        <maven.build.cache.exclude.subfolder>**/excluded_subfolder/**</maven.build.cache.exclude.subfolder>
+        <!-- Exclude by filepath + filename -->
+        <maven.build.cache.exclude.filepath>folder_outside_src/this_one_should_NOT_be_scanned.txt</maven.build.cache.exclude.filepath>
```

Comment:
It will be good to cove ambiguity cases here:

1. just `<maven.build.cache.exclude.subfolder>excluded_subfolder</maven.build.cache.exclude.subfolder>` - previously, I expected to have the whole dir excluded consistently with `second_folder_outside_src`. Now - it should exclude only a file with this name.
2. just `<maven.build.cache.exclude.subfolder>excluded_subfolder/**</maven.build.cache.exclude.subfolder>` - it should exclude `<dirRoot>/excluded_subfolder/**`, but not `<dirRoot>/subfolder/excluded_subfolder/**`
3. `<maven.build.cache.exclude.subfolder>/tmp/**</maven.build.cache.exclude.subfolder>` - I believe it should exclude the absolute path, e.g. all tmp files